### ISSUE TITLE
Feature Request: Ignore @covers option

### ISSUE THREAD

**Issue by BackEndTea** [sebastianbergmann/php-code-coverage#571]
Although the use of  ```@covers``` and ```@coversNothing``` is really useful for generating good coverage reports, there are also times when we want to ignore those settings. For example, i may wish to see exactly what my integration tests are covering.

My proposal would be to add a command line flag to phpunit, for example ```--ignore-covers``` or something like that. Which would cause ```@covers``` and ```@coversNothing``` to be ignored when calculating the code coverage.

Slightly related to: https://github.com/infection/infection/issues/88#issuecomment-351005953

**Comment by fkupper** [comment]
Just came up to the exact same need.

**Comment by Danon** [comment]
I would like to have this option as well, to enable/disable the `@covers` dynamically, depending on needs.

**Comment by Danon** [comment]
You can install tool `rawr/covers-ignore`, which will remove `@covers` and `@coversNothing` annotations from a test suite, which you can then run in `phpunit`.

Here's an example of usage in Github Actions: https://github.com/T-Regx/T-Regx/runs/7260340659?check_suite_focus=true#step:8:1

**Comment by BrianHenryIE** [comment]
I've written a script to match the changed lines in a `git diff` with the covered lines in code coverage output, and run the tests that cover the changed lines.

I use `@covers` everywhere, so unfortunately I'm only getting the tests that are written to test specifically those changed lines. It would be more effective if I could output a second codecoverage report which ignores the `@covers` annotations and gives me the true picture of every test that tested each line.

I'll try #573

**Comment by sebastianbergmann** [comment]
I do not think this would be useful. The purpose of the `#[Covers*]` attributes is to ensure that code coverage is only collected for the code unit(s) that a test is intended to cover. Ignoring these attributes would undermine the very purpose of having them in the first place.

I will not implement this.

**Comment by danon** [comment]
@sebastianbergmann We like using `#[Covers*]` because of the precision it gives us. Obviously using `--ignore-covers` always would defeat the purpuse; but we don't want to disable it **always**, just sometimes, on demand. Sometimes we'd like to run coverage for the sake of normal development, while using `#[Covers*]`. Other times, we'd like to see a "vanilla-coverage" without `#[Covers*]` scoping. Right now there's no way to do it, other than manually remove `#[Covers*]` annotations from codebase.

@sebastianbergmann Do you think you could sleep on that decision some more and take into account this circumstance?

**Comment by sebastianbergmann** [comment]
I communicated poorly, my bad. I meant to say that I do not believe this should be implemented in this library.

Have a look at https://github.com/sebastianbergmann/phpunit/pull/6598 for what I have in mind.

**Comment by danon** [comment]
@sebastianbergmann So the plan is to add `--disable-coverage-targetting`, which gives a similar outcome?

**Comment by sebastianbergmann** [comment]
Yes.
